File: python/gamex/platform_system.py

from __future__ import annotations
import os
from openstk.sfx.sys import ISystemSfx
from gamex.platform import AudioBuilderBase, AudioManager

# SystemAudioBuilder
class SystemAudioBuilder(AudioBuilderBase):
    def createAudio(self, path: object) -> object: raise NotImplementedError()
    def deleteAudio(self, audio: object) -> None: raise NotImplementedError()

# SystemSfx
class SystemSfx(ISystemSfx):
    source: PakFile
    audioManager: AudioManager

    def __init__(self, source: PakFile):
        self.source = source
        self.audioManager = AudioManager(source, SystemAudioBuilder())

    def createAudio(self, path: object) -> int: return self.audioManager.createAudio(path)[0]

File: dotnet/Base/GameX.Base/Platform_System.cs
using OpenStack.Sfx;
using System;

namespace GameX.Platforms
{
    /// <summary>
    /// SystemAudioBuilder
    /// </summary>
    public class SystemAudioBuilder : AudioBuilderBase<object>
    {
        public override object CreateAudio(object path) => throw new NotImplementedException();
        public override void DeleteAudio(object audio) => throw new NotImplementedException();
    }

    /// <summary>
    /// SystemSfx
    /// </summary>
    public class SystemSfx : ISystemSfx
    {
        readonly PakFile _source;
        readonly AudioManager<object> _audioManager;

        public SystemSfx(PakFile source)
        {
            _source = source;
            _audioManager = new AudioManager<object>(source, new SystemAudioBuilder());
        }

        public PakFile Source => _source;
        public IAudioManager<object> AudioManager => _audioManager;
        public object CreateAudio(object path) => _audioManager.CreateAudio(path).aud;
    }
}